generator/requirements.txt

```diff
@@ -0,0 +1,3 @@
+Faker==2.0.3
```

Comment:
If you wanted to generate a requirements.txt file with all your currently installed packages executing the following command from within the container is nice.

```
pip freeze >> requirements.txt
```

For this project, the result would be:

```
attrs==19.3.0
coverage==4.5.4
Faker==2.0.3
importlib-metadata==1.3.0
more-itertools==8.0.2
numpy==1.17.4
packaging==19.2
pandas==0.25.3
pluggy==0.13.1
py==1.8.0
pyparsing==2.4.5
pytest==5.3.1
pytest-cov==2.8.1
python-dateutil==2.8.1
pytz==2019.3
six==1.13.0
text-unidecode==1.3
wcwidth==0.1.7
zipp==0.6.0
```

The only downside I've run into is when you pip install a package in the editable mode (`pip install -e .`) the version of your package points to a directory.  For example, if the package `junk` was installed in editable mode the file would look like this:

```
attrs==19.3.0
coverage==4.5.4
Faker==2.0.3
importlib-metadata==1.3.0
# Editable Git install with no remote (junk==0.1.0)
-e /usr/src/junk
more-itertools==8.0.2
numpy==1.17.4
packaging==19.2
pandas==0.25.3
pluggy==0.13.1
py==1.8.0
pyparsing==2.4.5
pytest==5.3.1
pytest-cov==2.8.1
python-dateutil==2.8.1
pytz==2019.3
six==1.13.0
text-unidecode==1.3
wcwidth==0.1.7
zipp==0.6.0
```

Reply:
Hi Timothy, I left these out since they come as dependencies for the packages already listed in the requirements.txt. Is it standard to list all packages even those that are already baked into the dependencies?

Reply:
There are no hard rules, but usually, all the dependencies are in the requirements.txt file.

Here are some good reasons to add them:
* There is no question about what the configuration was when the code was "working".
* Dependencies could be added in your docker file that are not in the requirements.txt file.  Think about if we used the Anaconda image and only put packages in the requirements.txt file that weren't in Anaconda.  We could call numpy, pandas, matplotlib, ect... and the configuration would not be defined.

If we turned the project into a Python package we could also do select dependencies in the `setup.py` file.  One example would be if we wanted to exclude pytest from the normal distribution.  Other examples of selective dependencies that can be nice are build tools, database packages, jupyter, and ploting packages.